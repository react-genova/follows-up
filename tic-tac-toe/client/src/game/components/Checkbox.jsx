import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = ({ value, text, size, onChange, style }) => (
    <div style={{ display: 'flex', fontFamily: 'Arial, Helvetica, sans-serif', color: '#999', cursor: 'pointer', fontSize: size, ...style }} onClick={onChange}>
        <div style={{ border: '0.05em solid #999', borderRadius: '0.25em', width: '1em', height: '1em', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
            <div style={{ width: '0.85em', height: '0.85em', borderRadius: '0.25em', backgroundColor: value ? '#999' : 'transparent' }}/>
        </div>
        <span style={{ paddingLeft: text ? '0.25em' : '0'}}>{text}</span>
    </div>
);

Checkbox.propTypes = {
    value: PropTypes.bool,
    text: PropTypes.string,
    onChange: PropTypes.func,
    size: PropTypes.string,
    style: PropTypes.object,
};

Checkbox.defaultProps = {
    value: false,
    text: '',
    onChange: undefined,
    size: '1em',
    style: {},
};

export default Checkbox;
