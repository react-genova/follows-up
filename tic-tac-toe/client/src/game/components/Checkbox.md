### Checked checkbox

```jsx
<Checkbox value text="I'm checked" onChange={() => null} />
```

### Unchecked checkbox

```jsx
<Checkbox value={false} text="I'm not checked" onChange={() => null} />
```

### Checked big checkbox

```jsx
<Checkbox value size="2em" text="I'm checked" onChange={() => null} />
```

### Interactive checkbox

```jsx
const { useState } = require('react');
const MyComp = () => {
    const [value, setValue] = useState(false);
    return <Checkbox value={value} text="I'm checkable" onChange={e => setValue(!value)} />
}
<MyComp />
```