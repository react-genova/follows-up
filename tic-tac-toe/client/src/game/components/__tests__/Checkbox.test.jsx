import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import Checkbox from '../Checkbox';

describe('Checkbox component', () => {
    afterEach(cleanup);

    it('renders without crashing (unchecked)', () => {
        const { container } = render(<Checkbox text="click me" onChange={() => null} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders without crashing (without text)', () => {
        const { container } = render(<Checkbox onChange={() => null} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders without crashing (checked)', () => {
        const { container } = render(<Checkbox value text="click me" onChange={() => null} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
