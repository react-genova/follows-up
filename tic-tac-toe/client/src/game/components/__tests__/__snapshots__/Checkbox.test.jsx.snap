// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component renders without crashing (checked) 1`] = `
<div
  style="display: flex; font-family: Arial, Helvetica, sans-serif; color: rgb(153, 153, 153); cursor: pointer; font-size: 1em;"
>
  <div
    style="border: 0.05em solid #999; border-radius: 0.25em; width: 1em; height: 1em; display: flex; justify-content: center; align-items: center;"
  >
    <div
      style="width: 0.85em; height: 0.85em; border-radius: 0.25em; background-color: rgb(153, 153, 153);"
    />
  </div>
  <span
    style="padding-left: 0.25em;"
  >
    click me
  </span>
</div>
`;

exports[`Checkbox component renders without crashing (unchecked) 1`] = `
<div
  style="display: flex; font-family: Arial, Helvetica, sans-serif; color: rgb(153, 153, 153); cursor: pointer; font-size: 1em;"
>
  <div
    style="border: 0.05em solid #999; border-radius: 0.25em; width: 1em; height: 1em; display: flex; justify-content: center; align-items: center;"
  >
    <div
      style="width: 0.85em; height: 0.85em; border-radius: 0.25em; background-color: transparent;"
    />
  </div>
  <span
    style="padding-left: 0.25em;"
  >
    click me
  </span>
</div>
`;

exports[`Checkbox component renders without crashing (without text) 1`] = `
<div
  style="display: flex; font-family: Arial, Helvetica, sans-serif; color: rgb(153, 153, 153); cursor: pointer; font-size: 1em;"
>
  <div
    style="border: 0.05em solid #999; border-radius: 0.25em; width: 1em; height: 1em; display: flex; justify-content: center; align-items: center;"
  >
    <div
      style="width: 0.85em; height: 0.85em; border-radius: 0.25em; background-color: transparent;"
    />
  </div>
  <span
    style="padding-left: 0px;"
  />
</div>
`;
